pkg/api/v20240812preview/openshiftcluster_validatestatic.go

```diff
@@ -412,3 +423,60 @@ func (sv openShiftClusterStaticValidator) validateDelta(oc, current *OpenShiftCl
 
 	return nil
 }
+
+func (sv openShiftClusterStaticValidator) validatePlatformWorkloadIdentityProfile(path string, pwip *PlatformWorkloadIdentityProfile) error {
+	// PlatformWorkloadIdentityProfile being empty is acceptable
+	if pwip == nil {
+		return nil
+	}
+
+	// Validate the PlatformWorkloadIdentities
+	for n, p := range pwip.PlatformWorkloadIdentities {
+		if !stringutils.IsResourceIDFormatted(p.ResourceID) {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, fmt.Sprintf("%s.PlatformWorkloadIdentities[%d].resourceID", path, n), "ResourceID %s formatted incorrectly.", p.ResourceID)
+		}
+
+		if p.OperatorName == "" {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, fmt.Sprintf("%s.PlatformWorkloadIdentities[%d].resourceID", path, n), "Operator name is empty.")
+		}
+
+		if stringutils.ResourceType(p.ResourceID) != "userAssignedIdentities" {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, fmt.Sprintf("%s.PlatformWorkloadIdentities[%d].resourceID", path, n), "Resource must be a user assigned identity.")
+		}
+	}
+	return nil
+}
+
+func (sv openShiftClusterStaticValidator) validateIdentityXORServiceCredentials(pwip *PlatformWorkloadIdentityProfile, spPath string, spp *ServicePrincipalProfile) error {
+	// nothing to validate if the platform workload identity profile is nil
+	if pwip == nil {
+		if spp == nil {
+			return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, spPath, "Must provide either an identity or service principal credentials.")
+		}
+		return nil
+	}
+
+	if spp != nil && (spp.ClientID != "" || spp.ClientSecret != "") {
+		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, spPath, "Cannot use identities and service principal credentials at the same time.")
+	}
+
+	return nil
```

Comment:
This might be easier to understand that this is XOR, isn't it?

```suggestion
	if pwip == nil && spp == nil {
		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, spPath, "Must provide either an identity or service principal credentials.")
	}

	if pwip != nil && spp != nil && (spp.ClientID != "" || spp.ClientSecret != "") {
		return api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidParameter, spPath, "Cannot use identities and service principal credentials at the same time.")
	}

	return nil
```